### ISSUE TITLE
Getting 403 errors on some of the POST requests

### ISSUE THREAD

**Issue by pbteja1998** [serverless-nextjs/serverless-next.js#454]
**Describe the bug**
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<HTML><HEAD><META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">
<TITLE>ERROR: The request could not be satisfied</TITLE>
</HEAD><BODY>
<H1>403 ERROR</H1>
<H2>The request could not be satisfied.</H2>
<HR noshade size="1px">
This distribution is not configured to allow the HTTP request method that was used for this request. The distribution supports only cachable requests.
We can't connect to the server for this app or website at this time. There might be too much traffic or a configuration error. Try again later, or contact the app or website owner.
<BR clear="all">
If you provide content to customers through CloudFront, you can find steps to troubleshoot and help prevent this error by reviewing the CloudFront documentation.
<BR clear="all">
<HR noshade size="1px">
<PRE>
Generated by cloudfront (CloudFront)
Request ID: S9Bit4WZWnEar0Bo6NX7h-QbmHFUrnBF79lUp2LDYiLFoVPbw_bkTg==
</PRE>
<ADDRESS>
</ADDRESS>
</BODY></HTML>

I have suddenly started getting 403 error on some of the POST requests in my application.

My serverless.yml file for reference.
```
myNextApplication:
  component: serverless-next.js@1.14.0
  inputs:
    domain:
      - ${env.subdomain}
      - ${env.domain}
  build:
    cmd: yarn build
```

**Comment by danielcondemarin** [comment]
@pbteja1998 Is it an HTTP Post to a next API route?

**Comment by pbteja1998** [comment]
@danielcondemarin No, the POST request is to a backend Node Server.

**Comment by pbteja1998** [comment]
Everything was working till yesterday

**Comment by danielcondemarin** [comment]
> @danielcondemarin No, the POST request is to a backend Node Server.

Where do you host this backend node server? I don't see how it has anything to do with serverless-next.js unless I'm missing something,

**Comment by pbteja1998** [comment]
> > @danielcondemarin No, the POST request is to a backend Node Server.
> 
> Where do you host this backend node server? I don't see how it has anything to do with serverless-next.js unless I'm missing something,

It's hosted on AWS Lightsail. I don't know if it has anything to do with Serverless-NextJS. But the error says that there is some cloud front error. Hence opened an issue. My backend has no relation with cloud front.

**Comment by jaypeng2015** [comment]
The issue is because of the cloudfront settings:
The following source code shows  `allowedHttpMethods: ["HEAD", "GET"]` cannot be overwritten.
```
const cloudFrontOutputs = await cloudFront({
      defaults: {
        ttl: 0,
        ...cloudFrontDefaults,
        forward: {
          cookies: "all",
          queryString: true,
          ...cloudFrontDefaults.forward
        },
        // everything after here cant be overridden
        allowedHttpMethods: ["HEAD", "GET"],
        "lambda@edge": {
          ...defaultLambdaAtEdgeConfig,
          "origin-request": `${defaultEdgeLambdaOutputs.arn}:${defaultEdgeLambdaPublishOutputs.version}`
        },
        compress: true
      },
      origins: cloudFrontOrigins,
      ...(cloudFrontPriceClassInputs && {
        priceClass: cloudFrontPriceClassInputs
      })
    });

```

**Comment by dphang** [comment]
Thanks @jaypeng2015, looks like that is likely the problem

@pbteja1998 Please try to change the default behavior (*) to allow all HTTP methods and see if that works (just as testing, not sure if this is secure yet):

<img width="679" alt="Screen Shot 2020-09-08 at 11 06 02 PM" src="https://user-images.githubusercontent.com/3221524/92560831-e05c1980-f227-11ea-9b16-c407e7c344c8.png">

It seems like a limitation on CloudFront, it does not allow just GET, HEAD, POST, you must allow all methods: https://docs.aws.amazon.com/cloudfront/latest/APIReference/API_AllowedMethods.html

I think the fix might be to allow all HTTP methods and then setup permissions in the S3 origin so that only GET is supported. Not sure if other changes are needed in origin request handler for security reesons. I believe the bucket policy is already setup to only allow GET, so it might work to just do this.

```
{
    "Version": "2012-10-17",
    "Id": "PolicyForCloudFrontPrivateContent",
    "Statement": [
        {
            "Sid": " Grant a CloudFront Origin Identity access to support private content",
            "Effect": "Allow",
            "Principal": {
                "AWS": "arn:aws:iam::cloudfront:user/CloudFront Origin Access Identity E1L85R3YH128X4"
            },
            "Action": "s3:GetObject",
            "Resource": "arn:aws:s3:::nextjs-repros/*"
        }
    ]
}
```

@danielcondemarin are there any security concerns with just allowing all HTTP methods to be forwarded to the origin? Do we need to explicitly disallow certain HTTP methods by adding code in the Lambda origin request handler?

**Comment by dphang** [comment]
Released in latest 1.17 alpha.15 version
